### DISCUSSION TITLE
Shipping Fortigate Logs to Distributed SO 2.4.20

### DISCUSSION THREAD

**Discussion by NexusChak** [Security-Onion-Solutions/securityonion#11617] [Unsupported Versions]
Was reading through conversation between @dougburks and @bdavedu on #11321. As I don't quite get the concept behind SO's firewall settings, I'm having a hard time setting this up as well. My current setup is a v2.4.20 distributed cluster with a ManagerSearch node and Forward node (which I believe it's called sensor?)

Background Information:

Both ManagerSearch node and sensor tie to the same policy, which has Fortinet-Fortigate integration setup on port 9004.
Fortigate log is now being sent to sensor node via TCP syslog on port 9004.
tcpdump shows there are logs arriving sensor node.
On SO's firewall settings, I have configured "customhostgroup1" and "customportgroup1" with the Fortigate firewall IP and port 9004.
Under SO's firewall -> role -> sensor -> chain, I have set both DOCKER-USER and INPUT's customhostgroup1 -> portgroups to "customportgroup1.
For those external logs, like Fortigate logs in this case, should I have the syslog send to the ManagerSearch Node or Sensor, or I need to setup anything else? Just wondering am I doing anything wrong, and what would be the recommended way to do so.

How should I understand the SO's firewall settings? My understanding now is that logs are sending to sensor node, that's why I go to role -> sensor, and use cusomt hostgroup and port group to control the sensor node, and make it allow access from my Fortigate firewall to sensor node's port 9004. Is this a correct understanding?

Is there any other configuration needed to make logs flow from Sensor node and ManagerSearch node?

Is that going to be the same case for network traffic?

Any help will be appreciated. Thanks in advanced.

**Comment by bdavedu** [comment]
Hi, I hope it will help you.

SO Firewall Configuration:
From the Web UI of the OS we go to Administration -> Configuration -> Firewall. Once there, we first configure the source IP of the logs in "hostgroups". We select "customhostgroup1" and enter the IP of the forti (be careful since if there are VDOMs there may be IP routing so the IP that sent logs may be different from the one that acts as a gateway for the log source network.

Next, from "portgroups" we configure the destination port of the logs, in this case 9004. We configure it to "UDP" in "customportgroup1"

Finally, in the "role" tab we look for the role of our so-manager, in this case "managersearch". Then we go to chain -> DOCKER-USER -> hostgrpups -> customhostgroup1 -> portgroups-> and put customportgroup1. We do the same in chain -> INPUT.

Fortinet Fortigate logs integration configuration in elasticfleet:

From Kibana's Web UI we go to "Fleet". Then we go to "Agent Policies" and from there we create a copy of the "so-grid-nodes_general" policy. Then we go to "Agents" and assign the so-manager agent to the new policy created.

We click on the created policy and then click "Add integration". We search for "Fortinet FortiGate Firewall Logs" and install it. In the configuration we only enable logs via UDP on port 9004 and for IP 0.0.0.0.

Checks:
We can verify through TCP dump in the so-manager that packets arrive through port 9004.

**Reply by bdavedu** [reply]
To sum up, you have to send forti logs to Elastic Agent of you managersearch node and it will do rest of work.

**Reply by NexusChak** [reply]
@bdavedu I really appreciate all these detail information.

One last question, I have done basically the same setup but on the Foward node (the sensor?) instead of the manager node. Does it mean for logs, I'm supposed to use the manager node (which in my architecture, the ManagerSearch node) instead of the Forward node to install the Fortinet FortiGate Firewall Logs?

Also, if TCP dump could see packets coming from Fortigate with "A" ip, does it mean I could trust "A" ip will be the one that's being used for log shipping. And I should go ahead to put it in "customhostgroup1"?

**Reply by bdavedu** [reply]
First, Question--> I done this with managersearch node because I think it may be easer to carry fortilogs to logstash this way, but I don't know if it works with forward node.

Second Question--> Yes, you have too see which IP is sending logs and trust it. In my case, I have many VDOMs and I only want one of the VDOMs to send logs to security onion and it is difficult ro realize which IP is used as source because of internal NAT operations between VDOMs.

**Reply by NexusChak** [reply]
> First, Question--> I done this with managersearch node because I think it may be easer to carry fortilogs to logstash this way, but I don't know if it works with forward node.

Really hope a clarification from maintainers.

> Second Question--> Yes, you have too see which IP is sending logs and trust it. In my case, I have many VDOMs and I only want one of the VDOMs to send logs to security onion and it is difficult ro realize which IP is used as source because of internal NAT operations between VDOMs.
Gotcha. Thanks!

I will try to send Fortigate Logs to ManagerSearch Node instead and see if it works. Thanks again!

**Reply by NexusChak** [reply]
@bdavedu Just changed it to ManagerSearch. But still don't see any Fortigate Logs on Kibana yet. What keyword or data stream you used to filter out Fortigate logs?

Only differences:
1. I'm using TCP instead of UDP.
2. I'm using the original "so-grid-nodes_general" policy to add "Fortinet FortiGate Firewall Logs" integration instead of cloning a new policy then add integration.
3. Namespace "Default" (not sure if you are using default too)

But still, no log from Fortigate is received. At least I don't see any log under the data_stream.dataset: fortinet_fortigate.log

**Reply by bdavedu** [reply]
1. Try UDP
2. I don't see any problem here.
3. I am using the same namespace.

Try going to Stack Managment in Kibana and then go to Data Views and try to create new View. In the picture below you can see Index Patern name.

![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/129211552/d56beddd-aec7-4dbb-a873-323a71ba5c3c)

After thar go to Discover tab in Kibana and search the view you just created "forti_logs"

**Reply by NexusChak** [reply]
Just created a data view same as the one in picture. But still don't see any log from the Discover tab for this data view.
Last thing is to try with UDP. But I don't think this would make a difference.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/38964314/93f558a9-c1d1-4714-910b-9f14cf5e51b4)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/38964314/8cdf3c79-db3d-418f-a527-d471bd074223)

**Reply by bdavedu** [reply]
Are you sure about IP of forti that sends logs ?
Try to check in Kibana -> Stack Management -> Index Management -> Indices if you receive data.

![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/129211552/6a1fab78-6ada-42c9-a77f-953226fdfc70)

**Reply by NexusChak** [reply]
I did a TCP dump on the ManagerSearch. And I could see packets coming from the firewall IP to port 9004.

Just found one fortinet index, it should be created on the day I first have the integration installed. The size of the index is 249b, which means it's probably an empty index.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/38964314/7c35dfc2-24a0-420a-8331-27f493a85502)

How about index template? Would you mind to show me all your Fortinet related index templates? Although I already have the deprecated Fortinet integration uninstalled, but I'm suspecting it messed up the settings.

**Reply by bdavedu** [reply]
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/129211552/06251bb5-7409-427b-9d05-cc50ffa97130)

**Reply by NexusChak** [reply]
Just found that I have a bunch of same index templates for Fortinet with a prefix "so-", and they all have priority 501. Trying to remove them and see if this is the problem.

By the way, for the Fortinet data stream, would you show me which index template you are applying?

**Reply by bdavedu** [reply]
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/129211552/459ea8f5-4c3a-4b0a-b5df-1821c67d57aa)

**Reply by NexusChak** [reply]
Thanks. Everything should be identical now besides TCP vs UDP. But still no luck yet. No Fortinet index being created yet.

**Reply by bdavedu** [reply]
Try another fast deployment, standalone. May be there are something corrupt. I have to say that I am at 2.4.10 currently.

**Reply by jaypfe** [reply]
I have had problems with that integration. 

Try the Fortinet integration instead and configure as you have done here. That solved my problem.

**Reply by NexusChak** [reply]
@bdavedu Hmmmmm the Fortinet integration should not be supported at 2.4.10. Weird.

@jaypfe You mean the deprecated one?

**Reply by NexusChak** [reply]
@jaypfe Just had the deprecated one installed. And it just worked, now I'm getting the firewall logs. But they are not parsed. Is it supposed to be unparsed?

I believe something might went wrong with SO 2.4.20 and the new Fortinet integration.

**Reply by bdavedu** [reply]
![imagen](https://github.com/Security-Onion-Solutions/securityonion/assets/129211552/522e9c5d-4181-4a7d-8bae-48d81de77512)
I use this integration on 2.4.10.

**Reply by jaypfe** [reply]
So I used the new one that you are showing there but could never get any logs to actually come through. 

I could cheat it if I modified the index to bypass the fleet.final pipeline. I think I could arrive at the problem with enough messing around. But in the mean time I used the deprecated one and configured the fortigate area of the integration. Logs started coming in, and they were parsed just like you would expect. 

@JustAEngineer  can you drop your config here please? 

Also, you can make any integration work, you just need to build the index template. I have had discussions out here before where we worked through that back on .10. I also same the same exact discussion you posted already out there. But if you do a search for fortinet, or fortigate in the 2.4 section you will see them.

**Reply by NexusChak** [reply]
Just some updates, after removing all Forinet related components (including data stream, index template and its component templates and data views, etc). Then delete and install the latest version of Fortigate integration worked for me now. @jaypfe You may wanna give it a try.

But now I came to a new question, the default SO index template (so-logs) has a priority of 225, which is higher than the Fortigate integration created template (with priority 200), which means those Forigate indexes will only apply the "so-logs" template. This makes it difficult to manage and apply custom ILM policy. I have tried making the Fortigate template a priority of 250, it applies to the new coming index, but the status turned "Yellow" and no log is available to come in. Do you know what's happening here, and if it's recommended for all integrations to apply the "so-logs" template only?

**Reply by PacketAttack0** [reply]
This is my first go with SecOnion. Was really trying to have it act as my full log manager SIEM for all network devices starting with Fortinet devices. I have a clean Standalone install of 24.20 but still struggling to get the Fortigate logs to show. TCP dump shows SO is accepting UDP port 9004 syslog traffic from my Fortigate. Will work through the suggestions here and report back.
